src/views/Generator/TestRuleContainer/RulesDisabledOverlay.tsx

```diff
@@ -0,0 +1,29 @@
+import { useGeneratorStore } from '@/store/generator'
+import { Flex, Text } from '@radix-ui/themes'
+import { useTheme } from '@/hooks/useTheme'
+
+export function RulesDisabledOverlay() {
+  const theme = useTheme()
+  const rulesEnabled = useGeneratorStore((store) => store.rulesEnabled)
+
+  if (rulesEnabled) {
+    return null
+  }
+
+  return (
+    <Flex
```

Comment:
This doesn't prevent the user from using keyboard to edit the rules, but I'm not convinced we should disable/hide the UI if the switch is off - maybe keeping the rule list/rule editor accessible, but showing a very noticeable `Callout` component at the top of that pane will work better? Here's what it could look like (the copy is a quick draft)

<img width="859" alt="image" src="https://github.com/user-attachments/assets/2115abac-0011-4500-a615-65c7bb631fec" />
<img width="858" alt="image" src="https://github.com/user-attachments/assets/aec5fea0-4d1d-47d7-a050-b7397e96dae9" />

Might be worth discussing this with @majavojnoska

Reply:
Agree, this looks better 🙌 Replaced overlay with callout:

![screencapture 2025-02-19 at 14 46 38](https://github.com/user-attachments/assets/970470c2-d2a4-4b17-8fd8-afb159d0a40f)